eclass/kernel-install.eclass

```diff
@@ -660,13 +661,66 @@ kernel-install_extract_from_uki() {
 	local extract_type=${1}
 	local uki=${2}
 	local out=${3}
+	local out_temp=${T}/${extract_type}-section-dumped
 
 	# objcopy overwrites input if there is no output, dump the output in T.
 	# We unfortunately cannot use /dev/null here
 	$(tc-getOBJCOPY) "${uki}" "${T}/dump.efi" \
-		--dump-section ".${extract_type}=${out}" ||
-		die "Failed to extract ${extract_type}"
-	chmod 644 "${out}" || die
+		--dump-section ".${extract_type}=${out_temp}" ||
+			die "Failed to extract ${extract_type}"
+
+	# Sanity checks for kernel images
+	if [[ ${extract_type} == linux ]] &&
+		{ ! in_iuse secureboot || use secureboot ;}
+	then
+		# Extract the used SECUREBOOT_SIGN_CERT to verify the kernel image
+		local cert=${T}/pcrpkey
+		kernel-install_extract_from_uki pcrpkey "${uki}" "${cert}"
+		if [[ $(file "${cert}") == *"Certificate, Version=3" ]]; then
```

Comment:
I really dislike the idea of using `file(1)`, and especially in "human-readable text" mode that can change over time. Also, you're missing `IDEPEND` on it.

Reply:
How about using `head -n1` to check if the file starts with `-----BEGIN CERTIFICATE-----`, i.e. is a PEM format certificate?

Reply:
WFM.